app/generic/verifier/models.py

```diff
@@ -45,9 +42,8 @@ class PresentationExchange(BaseModel):
     connection_id: Optional[str] = None
     created_at: str
     proof_id: str
-    presentation: Optional[
-        Union[IndyProof, V20Pres, V20PresExRecordByFormat, Dict]
-    ] = None
+    presentation: Optional[Dict] = None
+    presentation_request: Optional[Dict] = None
```

Comment:
If we use the model types here we would also have typing in the swagger ui:

```suggestion
    presentation: Optional[IndyProof] = None
    presentation_request: Optional[IndyProofRequest] = None
```

Reply:
sure I'll give that a try. I think aca-py returns a dict so I'll have to construct that from the dict. will give it a go

Reply:
We do have to transform the dict to a model then in the v2 record_to_model  transformation